pkg/services/secrets/database.go

```diff
@@ -0,0 +1,78 @@
+package secrets
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/grafana/grafana/pkg/infra/log"
+	"github.com/grafana/grafana/pkg/services/secrets/types"
+	"github.com/grafana/grafana/pkg/services/sqlstore"
+)
+
+const dataKeysTable = "data_keys"
+
+var logger = log.New("secrets-store")
+
+func (s *SecretsService) GetDataKey(ctx context.Context, name string) (*types.DataKey, error) {
+	dataKey := &types.DataKey{}
+	var exists bool
+
+	err := s.sqlStore.WithDbSession(ctx, func(sess *sqlstore.DBSession) error {
+		var err error
+		exists, err = sess.Table(dataKeysTable).
+			Where("name = ? AND active = ?", name, s.sqlStore.Dialect.BooleanStr(true)).
+			Get(dataKey)
+		return err
+	})
+
+	if !exists {
+		return nil, types.ErrDataKeyNotFound
+	}
+
+	if err != nil {
+		logger.Error("Failed getting data key", "err", err, "name", name)
```

Comment:
@undef1nd should this really be logged at Error level? Looks to me like this will happen during normal operation whenever a new dak is created.

Reply:
When a new DEK is created, we check for `types.ErrDataKeyNotFound` in [this line](https://github.com/grafana/grafana/pull/39418/files#diff-13151734cb26fac3c6ae54024726f5a67801581ea6459e00fd848ea68d2800ccR88), so the rest error cased should be logged as Error, I guess 🤔 Please, correct me if I'm wrong.